jamazon/main.js

```diff
@@ -53,16 +54,96 @@ function renderCatalog(catalog) {
 
   for (let i = 0; i < catalog.items.length; i++) {
     let $cardCol = document.createElement('div')
-    $cardCol.classList.add('card-col', 'col-xl-2', 'col-lg-4', 'col-sm-6', 'col-12')
-    $cardCol.appendChild(render(catalog.items[i]))
+    $cardCol.classList.add('card-col', 'col-xl-2', 'col-lg-4', 'col-sm-12', 'col-12')
+    $cardCol.appendChild(card(catalog.items[i]))
     $itemRow.appendChild($cardCol)
   }
   return $container
 }
 
+function detailTemplate(item) {
+  let $detailCont = document.createElement('div')
+  let $detailImg = document.createElement('img')
+  let $detailName = document.createElement('h1')
+  let $detailBrand = document.createElement('h6')
+  let $detailDesc = document.createElement('p')
+  let $detailDet = document.createElement('p')
+  let $detailPrice = document.createElement('span')
+  let $detailInfo = document.createElement('div')
+  let $imgCont = document.createElement('div')
+  let $detailRow = document.createElement('div')
+
+  $detailCont.classList.add('container')
+  $detailInfo.classList.add('col', 'col-lg-9')
+  $detailRow.classList.add('row')
+  $imgCont.classList.add('col', 'col-lg-3')
+
+  $detailImg.setAttribute('src', item.imageUrl)
+
+  $detailName.textContent = item.name
+  $detailBrand.textContent = `By ${item.brand}`
+  $detailDesc.textContent = item.description
+  $detailDet.textContent = item.details
+  $detailPrice.textConent = item.price
+
+  $detailImg.classList.add('detail-img', 'd-flex', 'justify-content-start')
+  $detailName.classList.add('d-flex', 'justify-content-start')
+
+  $detailCont.appendChild($detailRow)
+  $detailRow.appendChild($imgCont)
+  $imgCont.appendChild($detailImg)
+  $detailRow.appendChild($detailInfo)
+  $detailInfo.appendChild($detailName)
+  $detailInfo.appendChild($detailBrand)
+  $detailInfo.appendChild($detailDesc)
+  $detailInfo.appendChild($detailDet)
+  $detailInfo.appendChild($detailPrice)
+
+  return $detailCont
+}
+
+function eachDetail(itemId, catalog) {
+  for (let i = 0; i < catalog.items.length; i++) {
+    if (catalog.items[i].itemId === itemId) {
+      return catalog.items[i]
+    }
+  }
+}
+
+let $app = document.querySelector('[data-view]')
+
+$app.addEventListener('click', function (element) {
+  let item = element.target.parentNode.closest('.card')
+  let itemId = item.getAttribute('data-item-id')
```

Comment:
Be careful here. If there was no `item`, calling `getAttribute` on it will throw an error.